tests/routes/test_vespasearch.py

```diff
@@ -0,0 +1,743 @@
+import csv
+from datetime import datetime
+import json
+import random
+import time
+from io import StringIO
+from pathlib import Path
+from typing import Mapping, Optional, Sequence
+
+import pytest
+from sqlalchemy import update
+from sqlalchemy.orm import Session
+
+from cpr_data_access.models.search import SearchParameters
+
+from app.api.api_v1.routers import search
+
+from app.core.lookups import get_country_slug_from_country_code
+
+from app.data_migrations.taxonomy_cclw import get_cclw_taxonomy
+from app.data_migrations.taxonomy_unf3c import get_unf3c_taxonomy
+
+from app.db.models.app import Organisation
+from app.db.models.law_policy import Geography
+from app.db.models.law_policy.family import (
+    DocumentStatus,
+    EventStatus,
+    FamilyCategory,
+    Family,
+    FamilyDocument,
+    FamilyDocumentType,
+    FamilyEvent,
+    FamilyOrganisation,
+    Slug,
+    Variant,
+)
+from app.db.models.law_policy.metadata import (
+    FamilyMetadata,
+)
+from app.db.models.document.physical_document import (
+    LanguageSource,
+    PhysicalDocument,
+    PhysicalDocumentLanguage,
+)
+from app.initial_data import run_data_migrations
+
+SEARCH_ENDPOINT = "/api/v1/searches"
+CSV_DOWNLOAD_ENDPOINT = "/api/v1/searches/download-csv"
+FIXTURE_DIR = Path(__file__).parents[1] / "search_fixtures"
+VESPA_FAMILY_PATH = FIXTURE_DIR / "vespa_family_document.json"
+VESPA_DOCUMENT_PATH = FIXTURE_DIR / "vespa_document_passage.json"
+
+
+def _parse_id(schema, convert_to: Optional[str] = None) -> str:
+    schema_id = schema["id"].split("::")[-1]
+    if convert_to is None:
+        return schema_id
+    else:
+        id_parts = schema_id.split(".")
+    return f"{id_parts[0]}.{convert_to}.{id_parts[2]}.{id_parts[3]}"
+
+
+def _get_family_fixture(doc):
+    with open(VESPA_FAMILY_PATH, "r") as vf:
+        for family in json.load(vf):
+            if family["id"] == doc["fields"]["family_document_ref"]:
+                return family
+
+
+def _fixture_docs():
+    with open(VESPA_DOCUMENT_PATH, "r") as vd:
+        documents = json.load(vd)
+
+    for doc in documents:
+        family = _get_family_fixture(doc)
+        yield doc, family
+
+
+def _populate_search_db_families(db: Session) -> None:
+    run_data_migrations(db)
```

Comment:
Just to confirm this is running migrations against rds not the search database - if so naming?

Reply:
fixed